realworld/src/processes/articleProcesses.ts

```diff
@@ -11,66 +10,64 @@ import {
 	NewCommentPayload
 } from './interfaces';
 
-const startLoadingArticleCommand = commandFactory(({ path, payload: { slug } }) => {
-	return [
-		replace(path('article', slug, 'item'), undefined),
-		replace(path('article', slug, 'comments'), []),
-		replace(path('article', slug, 'isLoading'), true),
-		replace(path('article', slug, 'isLoaded'), false)
-	];
+const startLoadingArticleCommand = commandFactory(({ state, payload: { slug } }) => {
+	state.article[slug] = {
+		item: undefined,
+		comments: [],
+		newComment: '',
+		isLoading: true,
+		isLoaded: false
+	};
 });
 
-const loadArticleCommand = commandFactory<SlugPayload>(async ({ get, path, payload: { slug } }) => {
-	const token = get(path('session', 'token'));
+const loadArticleCommand = commandFactory<SlugPayload>(async ({ state, payload: { slug } }) => {
+	const token = state.session && state.session.token;
```

Comment:
how do you feel about update the TS version and using the optional chaining magic?

Reply:
👍 sure

Reply:
✔️